File: src/Horse.Jockey/Core/MessageCount.cs
using System;
using Horse.Jockey.Helpers;

namespace Horse.Jockey.Core;

internal class MessageCount
{
    public long UnixTime;
    public long Received;
    public long Sent;
    public long Respond;
    public long Error;
    public long Delivered;
    public long NotRouted;
    public long Timeout;

    public MessageCount()
    {
        UnixTime = DateTime.UtcNow.ToUnixSeconds();
    }
};

File: web/src/models/message-count.ts

export interface CountRecord {

    /** Unix Time in seconds */
    u: number;

    /** Received */
    r: number;

    /** Sent */
    s: number;

    /** Respond */
    rs: number;

    /** Error */
    e: number;

    /** Delivered */
    d: number;

    /** Not Routed/Found */
    nr: number;

    /** Timeout (Request, Message or Acknowledge) */
    to: number;
}


export interface MessageCount {

    /** Name */
    n: string;

    /** Count Record Data */
    d: CountRecord[];

    /** Labels, generated in client side */
    labels?: string[];
}